src/tests/ftest/nvme/object.py

```diff
@@ -128,13 +28,104 @@ def setUp(self):
+            container (TestContainer): container to write
+            record (int): record size to be written
+            array_size (int, optional): size of array value to be written
+        """
+        # update data_array_size to use array value type
+        if array_size:
+            container.data_array_size.update(array_size)
+        # update record qty
+        container.record_qty.update(record)
+        self.log.info("Record Size: %s", container.record_qty)
+
+        # write multiple objects
+        container.write_objects()
+
+    @staticmethod
+    def container_read(container, array_size=None):
+        """Read and verify the written data.
+
+        Args:
+            container (TestContainer): container to read
+            array_size (int, optional): size of array value to be written
+        """
+        # update data_array_size to use array value type
+        if array_size:
+            container.data_array_size.update(array_size)
+        # read written objects and verify
+        container.read_objects()
+
+    def test_runner(self, namespace, record_size, array_size, thread_per_size=4):
+        """Perform simultaneous writes of varying record size to a container.
+
+        Args:
+            namespace (str): location from which to read the pool parameters
+            record_size (list): list of different record sizes to be written
+            array_size (int): size of array value to be written
+            thread_per_size (int, optional): threads per rec size
+        """
+        # create a new pool
+        pool = self.get_pool(namespace=namespace, connect=False, dmg=self.get_dmg_command().copy())
+
+        # display available space before write
+        self.log.info("%s space before writes", str(pool))
+        pool.display_space()
+        pool.connect()
+
+        # create a new container
+        container = self.get_container(pool)
+        container.open()
+
+        # initialize thread managers
+        write_manager = ThreadManager(self.container_write, self.get_remaining_time() - 30)
+        read_manager = ThreadManager(self.container_read, self.get_remaining_time() - 30)
+
+        # create read/write threads.
+        for rec in record_size:
+            for _ in range(thread_per_size):
+                # create threads using single value type
+                write_manager.add(container=container, record=rec, array_size=array_size)
+                read_manager.add(container=container, array_size=None)
+
+                # create threads using array value type
+                write_manager.add(container=container, record=rec, array_size=None)
+                read_manager.add(container=container, array_size=array_size)
+
+        # Run the write threads and check for errors
+        write_result = write_manager.run()
+        for result in write_result:
+            if not result.passed:
+                self.errors.append(result.result)
+        if self.errors:
+            return
```

Comment:
This just returns and the caller will use `report_errors`. This is because we run threads within threads, so need to propagate the errors to the original caller. I can add a comment if needed.